src/util/CompositeAsyncHandler.ts

```diff
@@ -43,7 +46,8 @@ export class CompositeAsyncHandler<TIn, TOut> implements AsyncHandler<TIn, TOut>
     try {
       handler = await this.findHandler(input);
     } catch {
-      throw new Error('All handlers failed. This might be the consequence of calling handle before canHandle.');
+      this.logger.warn('All handlers failed. This might be the consequence of calling handle before canHandle.');
+      throw new Error('All handlers failed');
```

Comment:
Would not change the errors messages. Also noticed this in some other cases. There's no reason to hide this now just because we have logging.

Reply:
I did make those changes because I was trying to do something about the redundancy. In this case, the detail is low-level info for a programmer, not an end user, so I figured there was no need to be elaborate inside of the message.

That said, I have done most of these based on gut feeling; we might want to reuse/create some advice on what level to use when, with what message, etc.

Reply:
Ah didn't see there was a comment about this. Was just thinking that some people might take more time now to find the issue since this is something that should only happen when developing something new and should never happen at runtime 😄 .